src/Web/WorkerController.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * WorkerController class file.
+ *
+ * @author Petra Barus <petra.barus@gmail.com>
+ * @author Adinata <mail.dieend@gmail.com>
+ */
+
+namespace UrbanIndo\Yii2\Queue\Web;
+
+use Yii;
+
+/**
+ * WorkerController is a web controller class that fetches work from queue and
+ * then run the job.
+ * The motivation comes from the HHVM limitation for running PHP terminal script.
+ * 
+ * @author Petra Barus <petra.barus@gmail.com>
+ * @author Adinata <mail.dieend@gmail.com>
+ */
+class WorkerController extends \yii\web\Controller
+{
+    
+    /**
+     * @var boolean
+     */
+    public $enableCsrfValidation = false;
+    
+    /**
+     * The default queue to use when there is no queue param sent in the request.
+     * @var string|array
+     */
+    public $defaultQueue = 'queue';
+    
+    /**
+     * The name of the request param to get the name of the queue component to use.
+     * This will check the parameter of the POST first. If there is no value, then it
```

Comment:
Still differs with actual implementation. From the doc it will looks like this:
private function getComponentParamFromRequest()
    {
        $request = Yii::$app->getRequest();
        return $request->post($this->queueParam, $request->get($this->queueParam, $this->defaultQueue));
    }